architecture/core_concepts/routes.adoc

```diff
@@ -12,41 +12,86 @@ toc::[]
+are provided and supported.
 
 ifdef::openshift-enterprise,openshift-origin[]
 [NOTE]
 ====
-See the link:../../install_config/install/deploy_router.html[Installation and
+See the xref:../../install_config/install/deploy_router.adoc#[Installation and
 Configuration] guide for information on deploying a router.
 ====
 endif::[]
 
-OpenShift routers provide external host name mapping and load balancing
-to link:pods_and_services.html#services[services] over protocols that
+A router uses the service selector to find the service and the endpoints backing
+the service. Service-provided load balancing is bypassed and replaced with the
+router's own load balancing. Routers watch the cluster API and automatically
+update their own configuration according to any relevant changes to the IP
+addresses of pods that provide the service.  Routers may be containerized.
+Custom routers can be deployed to communicate modifications of API objects to
+an external routing solution.
+
+In order to reach a router in the first place, requests for host names must
+resolve via DNS to a router or set of routers. The suggested method is to define
+a cloud domain with a wildcard DNS entry pointing to a virtual IP (VIP) backed by
+multiple router instances on designated nodes.
+
+Routes can be
+xref:../../architecture/core_concepts/routes.adoc#router-sharding[sharded]
+among the set of routers. Administrators can set up sharding on a cluster
+wide basis and users can set up sharding for the namespace in their project.
+Sharding allows the operator to define multiple router groups. Each router
+in the group serves only a subset of traffic.
+
+Router VIP configuration is described in
+ifdef::openshift-enterprise,openshift-origin[]
+xref:../../admin_guide/high_availability.adoc#configuring-a-highly-available-routing-service[High Availability].
+endif::[]
+ifdef::openshift-dedicated[]
+the https://docs.openshift.com/enterprise/3.1/admin_guide/high_availability.html#configuring-a-highly-available-routing-service[{product-title} Enterprise Cluster Administration documentation].
+endif::[]
+Routes using names and addresses outside the cloud domain require
+iconfiguration of individual DNS entries.
+
+
+If routes use host names/addresses outside the cloud domain, you will
+need to configure the DNS entries individually.
+
+{product-title} routers provide external host name mapping and load balancing
+of xref:pods_and_services.adoc#services[service] end points over protocols that
 pass distinguishing information directly to the router; the host name
 must be present in the protocol in order for the router to determine
 where to send it.
 
 Router plug-ins assume they can bind to host ports 80 and 443. This is to allow
 external traffic to route to the host and subsequently through the router.
-Routers also assume that networking is configured such that it can access all
-pods in the cluster.
+Sets of routers are configured onto different nodes.  Routers also assume that
```

Comment:
"Sets of routers are configured onto different nodes." is a bit wishy-washy.  And I'm not sure what you mean by "sets of routers".  Are you trying to say that only one router can be on a node?  If so, say:

Because a router that is using host networking binds to ports on the host node, only one router can be on each node (unless the ports are changed).